### PR TITLE
Support for named node and element sets for gmsh and Abaqus

### PR DESCRIPTION
**Pull request by dbeurle**
- Fix unexpected case change for named sets, see #493
- Add support for different dimensions in physicalNames, see #353

### CODE REVIEW

File: meshio/_mesh.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def __init__(
         cell_data=None,
         field_data=None,
         node_sets=None,
+        element_sets=None,
```
Comment (dbeurle) [inline meshio/_mesh.py]: @nschloe Why isn't `node_sets={}` used here instead of `None`.  I'm not a great python programmer.

Comment (gdmcbain) [reply meshio/_mesh.py]: Mutable default arguments are a no-no in Python; essentially there's only one of them that gets shared by all calls to the function. For the classic explanation see:

https://effbot.org/zone/default-values.htm

**Comment by nschloe** [comment]
How does this fix #353?

File: meshio/_mesh.py (hunk 1 above)
Comment (dbeurle) [reply meshio/_mesh.py]: Ha the more you know - thanks!

**Comment by dbeurle** [comment]
> How does this fix #353?

It doesn't in its current state.  However, hopefully it will as both node and element sets have the same name and their equivalent in gmsh would have to accommodate multiple sets with different dimensions and the same name.

**Comment by nschloe** [comment]
Fair enough.

**Pull request merged**